### PR TITLE
Document failure bubbling in Akka Typed

### PR DESCRIPTION
**Pull request by johanandren**
Mostly questions so far

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/fault-tolerance.md [hunk 1]
```diff
@@ -38,3 +45,20 @@ Java
...
 
+## Bubbling exceptions up
+
+In some scenarios it may be useful to push the decision about what to do on a failure upwards in the Actor hierarchy
+(in untyped Akka Actors this is how it works by default) and let the parent actor handle what should happen on failures.
+
+// FIXME is it useful that we should introduce something for it perhaps?
+
+// IDEA 1: parent will have to use watch, and act on children terminating
+// problems: no idea what exception happened (maybe that is good, to not tightly couple parent and children)
```
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/fault-tolerance.md]: That could be the reason for that there is a `def failure: Throwable` in `Terminated`. Check if that isn't populated exactly for child failures.

File: akka-docs/src/main/paradox/typed/fault-tolerance.md [hunk 2]
```diff
@@ -38,3 +45,20 @@ Java
...
+
+// IDEA 2: provide a special supervision strategy that queries the parent, 
+// put's the actor in limbo state until the parent has decided
+// problems: maybe this is just complicating things and not actually so useful?
```
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/fault-tolerance.md]: that would pretty much mean that we would be back to that the supervision logic would run in the parent actor, with all the complicated stuff for implementing that if/when we create a new runtime

**Comment by patriknw** [comment]
Refs https://github.com/akka/akka/issues/22665

File: akka-docs/src/main/paradox/typed/fault-tolerance.md (hunk 1 above)
Comment (patriknw) [reply akka-docs/src/main/paradox/typed/fault-tolerance.md]: see also "bubbling up failures" in https://github.com/akka/akka-meta/issues/46

File: akka-docs/src/main/paradox/typed/fault-tolerance.md (hunk 2 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/typed/fault-tolerance.md]: Yeah, that's what I was thinking as well.

File: akka-docs/src/main/paradox/typed/fault-tolerance.md (hunk 1 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/typed/fault-tolerance.md]: Thanks, I missed that there actually was something more than a mention that we need to solve it in the meta issue.

**Comment by johanandren** [comment]
I'm not super happy with the sample, but I got stuck figuring something better out yesterday.

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/WatchSpec.scala [hunk 3]
```diff
@@ -38,42 +38,111 @@ class WatchSpec extends TestKit("WordSpec", WatchSpec.config)
...
+      }
+      val terminated = probe.expectMsgType[Failed].t
+      terminated.failure should ===(Some(ex)) // here we get the exception from the child
+    }
+    "fail the actor itself with DeathPact if it does not accept Terminated" in {
+      case class Failed(t: Terminated) // we need to wrap it as it is handled specially
+      val probe = TestProbe[Any]()
+      val ex = new TestException("boom")
+      val grossoBosso = spawn(Behaviors.deferred[Any] { ctx ⇒
+        val middleManagement = ctx.spawn(Behaviors.deferred[Any] { ctx ⇒
```
Comment (patriknw) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/WatchSpec.scala]: :-)

File: akka-docs/src/main/paradox/typed/fault-tolerance.md [hunk 4]
```diff
@@ -38,3 +45,22 @@ Java
...
+In some scenarios it may be useful to push the decision about what to do on a failure upwards in the Actor hierarchy
+ and let the parent actor handle what should happen on failures (in untyped Akka Actors this is how it works by default).
+
+For a parent to be notified when a child is terminated it has to `watch` the child. If the child was stopped because of
+a failure this will be included in the `Terminated` signal in the `failed` field.
+
+If the parent in turn does not handle the `Terminated` message it will itself fail with an `akka.actor.typed.DeathPactException`.
+This means that a hierarchy of actors can have a child failure bubble up making each actor on the way stop but informing the
+top-most parent that there was a failure and how to deal with it, however, the original exception that caused the failure
+will only be available to the immediate parent.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/fault-tolerance.md]: does that mean that if one for some reason want to bubble the original exception up another level it should be done by handling `Terminated` and re-throw the exception in the failed field?

perhaps obvious, but worth mentioning?

**Review by patriknw — APPROVED** [review]
LGTM, with a possible addition

**Pull request merged**